pkg/operator/resources/namespaced/apiserver.go

```diff
@@ -0,0 +1,124 @@
+/*
+Copyright 2018 The CDI Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package namespaced
+
+import (
+	appsv1 "k8s.io/api/apps/v1"
+	corev1 "k8s.io/api/core/v1"
+	rbacv1beta1 "k8s.io/api/rbac/v1beta1"
+	"k8s.io/apimachinery/pkg/util/intstr"
+)
+
+const (
+	apiServerRessouceName     = "cdi-apiserver"
+	extensionAPIResourceName  = "cdi-extension-apiserver-authentication"
+	extensionAPIConfigMapName = "extension-apiserver-authentication"
+)
+
+func createAPIServerResources(args *FactoryArgs) []Resource {
+	return []Resource{
+		createAPIServerServiceAccount(),
+		createAPIServerRoleBinding(args.Namespace),
+		createAPIServerRole(),
+		createExtensionAPIServerRoleBinding(args.Namespace),
+		createExtensionAPIServerRole(),
+		createAPIServerService(),
+		createAPIServerDeployment(args.DockerRepo, args.APIServerImage, args.DockerTag, args.Verbosity, args.PullPolicy),
+	}
+}
+
+func createAPIServerServiceAccount() *corev1.ServiceAccount {
+	return createServiceAccount(apiServerRessouceName)
+}
+
+func createAPIServerRoleBinding(serviceAccountNamespace string) *rbacv1beta1.RoleBinding {
+	return createRoleBinding(apiServerRessouceName, apiServerRessouceName, apiServerRessouceName, serviceAccountNamespace)
+}
+
+func createAPIServerRole() *rbacv1beta1.Role {
+	role := createRole(apiServerRessouceName)
+	role.Rules = []rbacv1beta1.PolicyRule{
+		{
+			APIGroups: []string{
+				"",
+			},
+			Resources: []string{
+				"secrets",
+				"configmaps",
+			},
+			Verbs: []string{
+				"get",
+				"create",
+			},
+		},
+	}
+	return role
+}
+
+func createExtensionAPIServerRoleBinding(serviceAccountNamespace string) *rbacv1beta1.RoleBinding {
+	roleBinding := createRoleBinding(
+		extensionAPIResourceName,
+		extensionAPIResourceName,
+		apiServerRessouceName,
+		serviceAccountNamespace,
+	)
+	roleBinding.Namespace = "kube-system"
```

Comment:
Should have a constant for this.  I'm hoping that with the operator we will adopt the 'cdi' namespace as our default rather than kube-system'.

Reply:
no this has to be kube-system.  need to read a specific configmap that lives there